BBallStatsTrack/src/com/example/bballstatstrack/activities/AddPlayersToTeamsActivity.java

```diff
@@ -0,0 +1,198 @@
+        }
+        else if( mAwayTeamFragment.equals( current ) )
+        {
+            mHomeButton.setEnabled( true );
+            mAwayButton.setEnabled( false );
+        }
+    }
+
+    private class FragmentButtonListener implements OnClickListener
+    {
+        Fragment mFragment;
+
+        public FragmentButtonListener( Fragment fragment )
+        {
+            mFragment = fragment;
+        }
+
+        @Override
+        public void onClick( View v )
+        {
+            if( mFragment.equals( getCurrentFragment() ) )
+            {
+                return;
+            }
+            replaceCurrentFragment( mFragment );
+            updateView( mFragment );
+        }
+    }
+
+    private class ProceedButtonListener implements OnClickListener
+    {
+        public boolean hasIncompleteTeamNames()
+        {
+            return mHomeName.isEmpty() || mAwayName.isEmpty();
+        }
+
+        public boolean hasInsufficientPlayers()
+        {
+            return mHomeTeam.size() < 5 || mAwayTeam.size() < 5;
+        }
+
+        @Override
+        public void onClick( View v )
+        {
+            mHomeTeam = mHomeTeamFragment.getPlayerList();
+            mHomeName = mHomeTeamFragment.getTeamName();
+            mAwayTeam = mAwayTeamFragment.getPlayerList();
+            mAwayName = mAwayTeamFragment.getTeamName();
+            if( hasInsufficientPlayers() )
+            {
+                Toast.makeText( AddPlayersToTeamsActivity.this, "Insuffecient players. Cannot proceed.",
+                        Toast.LENGTH_SHORT ).show();
+                return;
+            }
+            if( hasIncompleteTeamNames() )
+            {
+                Toast.makeText( AddPlayersToTeamsActivity.this, "Team names are mandatory. Cannot proceed.",
+                        Toast.LENGTH_SHORT ).show();
+                return;
+            }
+            Intent intent = new Intent( AddPlayersToTeamsActivity.this, GameActivity.class );
+            intent.putExtra( HOME_TEAM_NAME, mHomeName );
+            setIntentExtras( intent, mHomeTeam, HOME_TEAM_NUMBERS, HOME_TEAM_MEMBER_NAMES );
+            intent.putExtra( AWAY_TEAM_NAME, mAwayName );
+            setIntentExtras( intent, mAwayTeam, AWAY_TEAM_NUMBERS, AWAY_TEAM_MEMBER_NAMES );
+            finish();
+            startActivity( intent );
+        }
+
+        private void setIntentExtras( Intent intent, SparseArray< String > team, String extraTeamNumbersIdentifier,
+                String extraTeamNamesIdentifier )
+        {
+            int size = team.size();
+            int[] numbers = new int[size];
+            String[] names = new String[size];
+            for( int i = 0; i < size; i++ )
+            {
+                numbers[i] = team.keyAt( i );
+                names[i] = team.valueAt( i );
+            }
+            intent.putExtra( extraTeamNumbersIdentifier, numbers );
+            intent.putExtra( extraTeamNamesIdentifier, names );
```

Comment:
Creating a team object which will hold List of Players as object(playerNumber, steal, points, rebouns and etc) will also simplify this, it will also be a good documentation.